x/bsonx/bsoncore/tranquil_document.go

```diff
@@ -62,3 +66,91 @@ func (d Document) LookupCaseInsensitive(key string) (string, Value, error) {
 	}
 	return "", Value{}, ErrElementNotFound
 }
+
+func (d Document) LookupRecursiveReturnFirst(key string) (Value, error) {
+	length, rem, ok := ReadLength(d)
+	if !ok {
+		return Value{}, NewInsufficientBytesError(d, rem)
+	}
+
+	length -= 4
+
+	var elem Element
+	for length > 1 {
+		elem, rem, ok = ReadElement(rem)
```

Comment:
You are now an officially certified MongoDB driver programmer - creating "rem" and passing it along to the next function all over the place :) (When in Rome do as the Romans).